solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,28 +19,52 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
+//        numberOfFile +=1 ;
+//        InstantiatableEntity game = createEntity(InstantiatableEntity.Type.BUILDING,
+//                InstantiatableEntity.Status.DESPAWNED, numberOfFile);
+//        List<InstantiatableEntity> listEntitites = new ArrayList<>();
+//        Savable<InstantiatableEntity> mapSavedEntitites = createSavable("saveGame"+numberOfFile, listEntitites);
+//        System.out.println("Before serialization: " + game);
+//        String filename = filesDirectory.concat("saveGame"+numberOfFile+".bin");
+//        try {
+//            saveGame(filename, mapSavedEntitites);
+//            Savable loadedData = loadGame(filename);
+//            System.out.println("After deserialization: " + loadedData);
+//        } catch (SaveGameException ex) {
+//
+//        }
+
     }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
+             ObjectOutputStream out = new ObjectOutputStream(fos)) {
+            out.writeObject(gameState);
+        } catch (IOException ex) {
+            throw new SaveGameException("Stream reading is failed");
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public MapState<GameObject> loadGame(String filename) throws SaveGameException {
+            try (FileInputStream fis = new FileInputStream(filesDirectory + File.separator + filename);
+                 ObjectInputStream in = new ObjectInputStream(fis)) {
+                return (MapState<GameObject>) in.readObject();
+            } catch (IOException | ClassNotFoundException ex) {
+                throw new SaveGameException("Stream reading is failed");
+            }
```

Comment:
&#39;catch rcurly&#39; has incorrect indentation level 12, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1628596259>*